GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -525,17 +528,26 @@ private void OnNewFileCreated(string relativePath, bool isDirectory)
                 {
                     if (isDirectory)
                     {
-                        GitCommandLineParser gitCommand = new GitCommandLineParser(this.Context.Repository.GVFSLock.GetLockedGitCommand());
-                        if (gitCommand.IsValidGitCommand)
-                        {
-                            // TODO(Mac): Ensure that when git creates a folder all files\folders within that folder are written to disk
-                            EventMetadata metadata = this.CreateEventMetadata(relativePath);
-                            metadata.Add("isDirectory", isDirectory);
-                            this.Context.Tracer.RelatedWarning(metadata, $"{nameof(this.OnNewFileCreated)}: Git created a folder, currently an unsupported scenario on Mac");
-                        }
-                        else
+                        if (!relativePath.Equals(GVFSConstants.DotGit.Root, StringComparison.OrdinalIgnoreCase))
                         {
-                            this.FileSystemCallbacks.OnFolderCreated(relativePath);
+                            string lockedGitCommand = this.Context.Repository.GVFSLock.GetLockedGitCommand();
+                            GitCommandLineParser gitCommand = new GitCommandLineParser(lockedGitCommand);
+                            if (gitCommand.IsValidGitCommand)
+                            {
+                                EventMetadata metadata = this.CreateEventMetadata(relativePath);
+                                metadata.Add(nameof(lockedGitCommand), lockedGitCommand);
+                                metadata.Add(TracingConstants.MessageKey.InfoMessage, "Git command created new folder");
+                                this.Context.Tracer.RelatedEvent(EventLevel.Informational, $"{nameof(this.OnNewFileCreated)}_GitCreatedFolder", metadata);
+
+                                // Record this folder as expanded so that GitIndexProjection will re-expand the folder
+                                // when the projection change completes.  This is the closest we can get to converting this

```

Comment:
Why is this the closest we can get? Why can't we just mark the folder as empty again?

Reply:
I wasn't asking for an alternate approach, just pointing out that the comment is making a false claim :-). Marking the folder as expanded seems like a fine approach.

And yea, with the code as it is _today_, marking the folder empty will likely fail, because we will fail when creating a duplicate placeholder. But that will change soon with @nickgra 's PR for atomic placeholder creation.

Reply:
> I wasn't asking for an alternate approach, just pointing out that the comment is making a false claim :-).

Ok thanks :) I'll update this comment.